### PR TITLE
feat: Mandatory question indicator added

### LINKED ISSUE #885: [FormTribe 🔥][250 Points] Mandatory Question Indicator
### Welcome to the FormTribe!

This issue will get you **250 points**  for the [FormTribe Hackathon](https://formtribe.com)

**Preparation:**
If this is your first time contributing, please read:

- Our [introductory blog post](https://formbricks.com/blog/join-the-formtribe)
- Overview of [Code Guidelines at Formbricks.](https://formbricks.com/docs/contributing/how-we-code)

**Only when your PR follows _all_ of the best practices it will get merged.**

Want more points quick? Complete [Side Quests 🏰 ](https://formbricks.com/formtribe#side-quests)

-------
### Issue

Currently, we don't have a mandatory field indicator

### Describe the solution you'd like

Add one :)

image

As you can see, we're going with an inverted approach: Marking optional fields. We do this for a handful of reasons:

- In surveying, its a good practice to only ask what's really needed so by default a question is mandatory
- We use a text-based indicator for accessibility reasons (instead e.g. a red * )

### Tasks: 

1. Surveys are being rendered by the Formbricks survey package @formbricks/surveys packages. This is where you need to add this indicator. Please create a reusable component for all question types.

2. Please test in the preview as well as end-user facing surveys

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by gupta-piyush19**
## What does this PR do?
mandatory question indicator added

Fixes #885 

image
image

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [x] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change adds a new database migration
- [ ] This change requires a documentation update

## How should this be tested?

- Test A
- Test B

## Checklist

- [x] Added a screen recording or screenshots to this PR
- [ ] Filled out the "How to test" section in this PR
- [x] Read the [contributing guide](https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

**Comment by jobenjada** [comment]
legend! What a turnaround :D

File: packages/surveys/src/components/Headline.tsx [hunk 1]
```diff
@@ -1,16 +1,31 @@
 interface HeadlineProps {
   headline?: string;
   questionId: string;
+  labelStyle?: any;
```
Comment (jobenjada) [inline packages/surveys/src/components/Headline.tsx:4]: Hey @gupta-piyush19 

Thanks for picking this up!

The only open question is why introduced this prop if it is not used. What is / was the idea for it?

Other than that, ready to merge :)

**Review by jobenjada — CHANGES_REQUESTED** [review]
Please remove the prop labelStyle

File: packages/surveys/src/components/Headline.tsx (hunk 1 above)
Comment (gupta-piyush19) [reply packages/surveys/src/components/Headline.tsx]: @jobenjada , I added this to pass extra styles for thankyou card, but then used different approach. Removing it.

**Review by jobenjada — APPROVED** [review]

**Pull request merged**